### DISCUSSION TITLE
I'm building a project with Next.js Serverless functions. Will Prisma+PostgreSQL work well for that? Is there anything I should be aware of?

### DISCUSSION THREAD

**Discussion by lumenwrites** [prisma/prisma#9093] [Q&A] [answered]
Hi! I've recently discovered Next.js serverless functions, and they're amazing, I've decided to build my project using them.

I've talked to people on forums, and it seems like Prisma + PostgreSQL will be a good fit for my project (in terms of storing and managing data).

My only question is - does Prisma/PostgreSQL work well with serverless functions? 

Someone told me that it's not scalable, something to do with the number of database connections, since every time  a serverless function executes, there will be a new one.

I'm new at this stuff, so I don't really understand what that means or how big of an issue this is. Do you guys know if this works well? Are there any other potential issues I should be aware of? Can you recommend some good tutorials/examples about setting this up?

**Comment by ryands17** [comment, marked as answer]
@lumenwrites 👋 

You should be fine using Next.js serverless functions. The advantage of using Next.js in this case is that you just need the functions for writes as reads can be done using Next.js functions like `getServerSideProps`.

> Someone told me that it's not scalable, something to do with the number of database connections, since every time a serverless function executes, there will be a new one.

If the serverless function is re-used, then there shouldn't be a new connection. You would need to initialise `PrismaClient` outside the handler and it should be fine.

We already have examples of Next.js with Prisma [here](https://www.prisma.io/nextjs). Let me know if you have any other queries 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
